boot/freeldr/freeldr/arch/uefi/uefimem.c

```diff
@@ -0,0 +1,296 @@
+/*
+ * PROJECT:     Freeldr UEFI Extension
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     UEFI Memory Managment Functions
+ * COPYRIGHT:   Copyright 2022 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/**** Includes ***********************************************************************************/
+
+#include <uefildr.h>
+
+#include <debug.h>
+DBG_DEFAULT_CHANNEL(WARNING);
+
+#define NEXT_MEMORY_DESCRIPTOR(Descriptor, DescriptorSize) \
+    (EFI_MEMORY_DESCRIPTOR*)((char*)(Descriptor) + (DescriptorSize))
+#define EXIT_STACK_SIZE 0x1000
+#define UNUSED_MAX_DESCRIPTOR_COUNT 10000
+
+ULONG
+AddMemoryDescriptor(
+    IN OUT PFREELDR_MEMORY_DESCRIPTOR List,
+    IN ULONG MaxCount,
+    IN PFN_NUMBER BasePage,
+    IN PFN_NUMBER PageCount,
+    IN TYPE_OF_MEMORY MemoryType);
+
+/**** Global Variables ***************************************************************************/
+
+extern EFI_SYSTEM_TABLE * GlobalSystemTable;
+extern EFI_HANDLE GlobalImageHandle;
+extern REACTOS_INTERNAL_BGCONTEXT framebufferData;
+
+EFI_MEMORY_DESCRIPTOR* EfiMemoryMap = NULL;
+UINT32 FreeldrDescCount;
+
+PVOID OsLoaderBase;
+SIZE_T OsLoaderSize;
+EFI_HANDLE PublicBootHandle;
+PVOID ExitStack;
+PVOID EndofExitStack;
+
+/**** Private Functions **************************************************************************/
+VOID
+PUEFI_LoadMemoryMap(UINTN*  LocMapKey,
+                    UINTN*  LocMapSize,
+                    UINTN*  LocDescriptorSize,
+                    UINT32* LocDescriptorVersion)
```

Comment:
New functions --> use SAL annotations. Also if some of these functions are used only within that single file, make them `static`.

Reply:
lemme know if _Out_ seemed like it fitted to you as well!